core/lib/instrument_api.h

```diff
@@ -6228,11 +6228,14 @@ DR_API
  * \note Use \p size == 1 to flush fragments containing the instruction at address
  * \p start. A flush of \p size == 0 is not allowed.
  *
+ * \note Use flush_completion_callback to specify logic to be executed after the flush
+ * and before the threads are resumed.
+ *
  * \note As currently implemented, dr_delay_flush_region() with no completion callback
  * routine specified can be substantially more performant.
  */
 bool
-dr_flush_region(app_pc start, size_t size);
+dr_flush_region(app_pc start, size_t size, void (*flush_completion_callback)());
```

Comment:
suggest: Consider having a user_data param to the callback.  We've regretted not having them for all the DR event callbacks and in other places (xref ... some issue, will find it, plus internal issue ...).  It makes it nicer for the user to avoid global/static data.

Reply:
Done.